Velocity_Model/z.py

```diff
@@ -0,0 +1,173 @@
+"""
+Library functions for extracting Z values from the NZVM in a pythonic manner
+Supports NZVM versions 2.03, 2.04, 2.06
+"""
+import os
+from pathlib import Path
+import shutil
+import subprocess
+import tempfile
+
+import numpy as np
+import pandas as pd
+from matplotlib import path as mpltPath
+
+OUT_BASIN_SIGMA = 0.5
+
+IN_BASIN_SIGMA = 0.3
+
+file_output_format = {"Z1.0": "Z/Z_1.0.txt", "Z2.5": "Z/Z_2.5.txt"}
+column_names = {"Z1.0": "Z_1.0(km)", "Z2.5": "Z_2.5(km)"}
+
+
+def get_basin_full_path(basin_list):
+    vm_dir = os.path.abspath(
+        os.path.join(os.path.dirname(os.path.realpath(__file__)), "../")
+    )
+    return [os.path.join(vm_dir, basin) for basin in basin_list]
+
+
+v203_basin_list = get_basin_full_path(
+    [
+        "Data/NI_BASINS/Wellington_Polygon_Wainuiomata_WGS84.txt",
+        "Data/Boundaries/NewCanterburyBasinBoundary_WGS84_1m.txt",
+        "Data/Boundaries/BPVBoundary.txt",
+        "Data/SI_BASINS/Cheviot_Polygon_WGS84.txt",
+        "Data/SI_BASINS/Hanmer_Polygon_WGS84.txt",
+        "Data/SI_BASINS/Kaikoura_Polygon_WGS84.txt",
+        "Data/SI_BASINS/Marlborough_Polygon_WGS84_v0p1.txt",
+        "Data/SI_BASINS/Nelson_Polygon_WGS84.txt",
+        "Data/SI_BASINS/NorthCanterbury_Polygon_WGS84.txt",
+        "Data/Boundaries/WaikatoHaurakiBasinEdge_WGS84.txt",
+    ]
+)
+v204_basin_list = v203_basin_list + get_basin_full_path(
+    [
+        "Data/USER20_BASINS/WanakaOutlineWGS84.txt",
+        "Data/USER20_BASINS/WakatipuBasinOutlineWGS84.txt",
+        "Data/USER20_BASINS/alexandra_outline.txt",
+        "Data/USER20_BASINS/ranfurly_outline.txt",
+        "Data/USER20_BASINS/NE_otago/NE_otago_A_outline.txt",
+        "Data/USER20_BASINS/NE_otago/NE_otago_B_outline.txt",
+        "Data/USER20_BASINS/NE_otago/NE_otago_C_outline.txt",
+        "Data/USER20_BASINS/NE_otago/NE_otago_D_outline.txt",
+        "Data/USER20_BASINS/NE_otago/NE_otago_E_outline.txt",
+        "Data/USER20_BASINS/bal_outline_WGS84.txt",
+        "Data/USER20_BASINS/dun_outline_WGS84.txt",
+        "Data/USER20_BASINS/mos_outline_WGS84.txt",
+        "Data/USER20_BASINS/Murchison_Basin_Outline_v1_WGS84.txt",
+        "Data/USER20_BASINS/hakataramea_outline_WGS84.txt",
+        "Data/USER20_BASINS/waitaki_outline_WGS84.txt",
+    ]
+)
+v205_basin_list = v204_basin_list
+v206_basin_list = v205_basin_list + get_basin_full_path(
+    [
+        "Data/USER20_BASINS/SpringsJ_basin_outline_v1_WGS84.txt",
+        "Data/USER20_BASINS/Karamea_basin_outline_v1_WGS84.txt",
+        "Data/USER20_BASINS/CollingwoodBasinOutline_1_WGS84_v1.txt",
+        "Data/USER20_BASINS/CollingwoodBasinOutline_2_WGS84_v1.txt",
+        "Data/USER20_BASINS/CollingwoodBasinOutline_3_WGS84_v1.txt",
+    ]
+)
+basin_outlines = {
+    "2.03": v203_basin_list,
+    "2.04": v204_basin_list,
+    "2.05": v205_basin_list,
+    "2.06": v206_basin_list,
+}
```

Comment:
Maybe add a version for 1.66? (Or what ever the last 1.XX version is)
I like how you've used the incremental nature of the basins though.

Reply:
I thought about it, but we don't really use 1.66 anymore - if we did need it for 1.66 we can add it ad-hoc. I think basin stuff is mostly for NZVM 2+ .

It gets a little messier when we incrementally add newer versions of existing basins but not hugely messy.

Reply:
Yeah, updating basins was the main issue I had with the current style.
I have an idea about switching over to using dictionaries and replacing keys if we need to do that though.